### PR TITLE
Add support for PostgreSQL password expiration.

### LINKED ISSUE #42893: PostgreSQL password expiration
### Description of Issue/Question
PostgreSQL module and states have no support for "VALID UNTIL" clause, to make password expire.

### Setup
Not relevant.

### Steps to Reproduce Issue
Not relevant.

### Versions Report
Not relevant

### PR DESCRIPTION
**Pull request by gilbsgilbs**
### What does this PR do?
Add support for PostgreSQL password expiration.

### What issues does this PR fix or reference?
Closes #42893

### Previous Behavior
Unchanged

### New Behavior
Allow to set "valid_until" in Posgres module and states.

### Tests written?
Yes

### CODE REVIEW

**Comment by gilbsgilbs** [comment]
Note that it's not possible to set `VALID UNTIL` back to `NULL` once it has been set. If one want to drop `VALID UNTIL` clause, one will have to set `VALID UNTIL` value to string `infinity` (which is a valid PostgreSQL timestamp), just like do in PostgreSQL.

File: salt/modules/postgres.py [hunk 1]
```diff
@@ -886,7 +886,7 @@ def get_bool(rowdict, key):
         for date_key in ('expiry time',):
             try:
                 retrow[date_key] = datetime.datetime.strptime(
-                    row['date_key'], '%Y-%m-%d %H:%M:%S')
+                    row[date_key], '%Y-%m-%d %H:%M:%S')
```
Comment (cachedout) [inline salt/modules/postgres.py]: Is this a bug fix? I'm curious if the previous behavior was incorrect here.

Comment (gilbsgilbs) [inline salt/modules/postgres.py]: I assume this was a bug, yes. `expiry time` was set to `None` invariably.

Comment (gilbsgilbs) [inline salt/modules/postgres.py]: Just to be clearer, the reason for that is that the query above doesn't select any `"date key"` column. Meaning that this used to invariably raise a `KeyError` that is handled below.

I'm unsure why `KeyError` would still need to be caught though. `strptime` can only raise `ValueError` and according to the query, I think `expiry time` key will always exist. Let me know if you think `KeyError` is superfluous.

Comment (cachedout) [inline salt/modules/postgres.py]: I agree with your analysis here. IMHO, the `KeyError` handling should be safe to remove.

Comment (gilbsgilbs) [inline salt/modules/postgres.py]: Done with 18753597f7871fe00cd6039538576b2729d4bf99.

**Comment by cachedout** [comment]
@gilbsgilbs We failed two unit tests I'm afraid:

https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/13822/testReport/junit/unit.modules.test_postgres/PostgresTestCase/test_user_exists/

and

https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/13822/testReport/junit/unit.modules.test_postgres/PostgresTestCase/test_user_list/

**Comment by gilbsgilbs** [comment]
@cachedout Sorry about that. That should be fixed now. It was a problem with mocks. In one test, `expiry time` was mocked to `None` where it should have been empty string, therefore `strptime` raised a `TypeError` instead of a `ValueError`.  In the other test, it wasn't even mocked, wich caused a `KeyError` that used to be silented.

**Comment by cachedout** [comment]
@gilbsgilbs Yes, the build looks fine now. I will review this today.

**Review by cachedout — APPROVED** [review]

**Pull request merged**